lib/kotkowo_web/live/news_live/found_home.ex

```diff
@@ -0,0 +1,389 @@
+      tag: socket.assigns.tag,
+      castrated: socket.assigns.castrated,
+      ages: socket.assigns.ages,
+      colors: socket.assigns.colors,
+      genders: socket.assigns.genders,
+      date_from: socket.assigns.date_from,
+      date_to: socket.assigns.date_to,
+      page: socket.assigns.page,
+      limit: limit
+    }
+
+    socket =
+      socket
+      |> assign(:max_page, max_page)
+      |> push_patch(to: ~p"/aktualnosci/znalazly-dom?#{query_params}")
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_event("select_page", %{"value" => page}, socket) do
+    query_params = %{
+      name: socket.assigns.name,
+      tag: socket.assigns.tag,
+      castrated: socket.assigns.castrated,
+      ages: socket.assigns.ages,
+      genders: socket.assigns.genders,
+      date_from: socket.assigns.date_from,
+      colors: socket.assigns.colors,
+      date_to: socket.assigns.date_to,
+      page: page,
+      limit: socket.assigns.limit
+    }
+
+    socket = push_patch(socket, to: ~p"/aktualnosci/znalazly-dom?#{query_params}")
+    {:noreply, socket}
+  end
+
+  def query_to_bool(query) do
+    case query do
+      "on" -> true
+      "true" -> true
+      "false" -> false
+      "" -> nil
+      _ -> query
+    end
+  end
+
+  defp search_tags(cats, search) do
+    if String.trim(search) == "" do
+      cats
+    else
+      Enum.filter(cats, fn adopted_cat ->
+        search = String.downcase(search)
+
+        adopted_cat.cat.tags
+        |> Enum.map(&String.downcase/1)
+        |> Enum.any?(&String.contains?(&1, search))
+      end)
+    end
+  end
+
+  defp parse_date(date_str) when date_str == "", do: nil
+
+  defp parse_date(date_str) do
+    [year, month, day] =
+      date_str
+      |> String.split("-")
+      |> Enum.map(&String.to_integer/1)
+
+    Date.new!(year, month, day)
+  end
+
+  defp filter_dates(cats, "", ""), do: cats
+
+  defp filter_dates(cats, from, to) do
+    from = parse_date(from)
+    to = parse_date(to)
+
+    Enum.filter(cats, fn adopted_cat ->
+      adoption_date = DateTime.to_date(adopted_cat.adoption_date)
+
+      within_range =
+        (from == nil or Date.compare(adoption_date, from) in [:gt, :eq]) and
+          (to == nil or Date.compare(adoption_date, to) in [:lt, :eq])
+
+      true in [within_range]
+    end)
+  end
+
+  defp filter_colors(cats, colors) do
+    if Enum.empty?(colors) do
+      cats
+    else
+      Enum.filter(cats, fn adopted_cat ->
+        Atom.to_string(adopted_cat.cat.color.value) in colors
+      end)
+    end
+  end
+
+  defp filter_genders(cats, genders) do
+    if Enum.empty?(genders) do
+      cats
+    else
+      Enum.filter(cats, fn adopted_cat ->
+        Atom.to_string(adopted_cat.cat.sex.value) in genders
+      end)
+    end
+  end
+
+  defp filter_ages(cats, ages) do
+    if Enum.empty?(ages) do
+      cats
+    else
+      Enum.filter(cats, fn adopted_cat ->
+        Atom.to_string(adopted_cat.cat.age.value) in ages
```

Comment:
aren't ages already parsed to atoms?
this wouldn't work. no?

Reply:
ages from query params are strings, so it has to stay